packages/ui/src/viewer-table/DataTable/SchemaTable/Columns/AllowedValues.tsx

```diff
@@ -0,0 +1,260 @@
+	) {
+		handleCodeListsWithCountRestrictions(
+			restrictionsValue.codeList,
+			restrictionsValue.count,
+			restrictionItems,
+			schemaField.isArray || false,
+			schemaField.delimiter ?? ',',
+		);
+	}
+
+	if (schemaField.unique) {
+		restrictionItems.push({ prefix: null, content: 'Must be unique' });
+	}
+
+	if (restrictionItems.length === 0) {
+		restrictionItems.push({ prefix: null, content: 'None' });
+	}
+
+	return restrictionItems;
+};
+
+const handleDependsOn = (schemaRestrictions: SchemaRestrictions): restrictionItem[] => {
+	const restrictionItems: restrictionItem[] = [];
+
+	if (schemaRestrictions && 'compare' in schemaRestrictions && schemaRestrictions.compare) {
+	}
+	if (
+		schemaRestrictions &&
+		'if' in schemaRestrictions &&
+		schemaRestrictions.if &&
+		'conditions' in schemaRestrictions.if &&
+		schemaRestrictions.if.conditions &&
+		Array.isArray(schemaRestrictions.if.conditions)
+	) {
+		const allFields: string[] = [];
+		schemaRestrictions.if.conditions.forEach((condition) => {
+			if (condition.fields !== undefined) {
+				if (Array.isArray(condition.fields)) {
+					allFields.push(...condition.fields);
+				} else if (typeof condition.fields === 'string') {
+					allFields.push(condition.fields);
+				}
+			}
+		});
+		// Since there can be multiple fields with multiple conditions, there is a possibility that there are duplicates
+		const uniqueFields = [...new Set(allFields)];
+		if (uniqueFields.length > 0) {
+			restrictionItems.push({
+				prefix: 'Depends on:\n',
+				content: uniqueFields.join(', '),
+			});
+		}
+	}
+
+	return restrictionItems;
+};
+export const renderAllowedValuesColumn = (restrictions: CellContext<SchemaField, SchemaRestrictions>) => {
+	const restrictionItems = computeAllowedValuesColumn(restrictions);
+	const restrictionsValue: SchemaRestrictions = restrictions.getValue();
+
+	const theme = useThemeContext();
+
+	const linkStyle = (theme: Theme) => css`
+		${theme.typography?.label};
+		color: ${theme.colors.black};
+		cursor: pointer;
+		display: inline-flex;
+		align-items: center;
+		background: none;
+		border: none;
+		padding: 0;
+		margin-top: 4px;
+		text-decoration: underline;
+		&:hover {
+			text-decoration: underline;
+		}
+	`;
+
+	const pillStyle = {
+		fontFamily: 'B612 Mono',
+		color: theme.colors.accent_dark,
+		fontWeight: '400',
+		lineHeight: '20px',
+		fontSize: '13px',
+	};
+	//TODO: implement the modal
+	const handleViewDetails = () => {
+		alert('Modal has been opened\n\n\n Hello World');
+	};
+
+	const renderRestrictionItem = (item: restrictionItem) => {
+		const { prefix, content } = item;
+		if (prefix === 'Depends on:\n') {
+			return (
+				<div key={`${prefix}-${content}`}>
+					{prefix && <strong>{prefix}</strong>}
+					<Pill size="extra-small" style={pillStyle}>
+						{content}
+					</Pill>
+				</div>
+			);
+		}
+
+		return (
+			<div key={`${prefix}-${content}`}>
+				{prefix && <strong>{prefix}</strong>}
+				{content}
+			</div>
+		);
+	};
```

Comment:
using this in a map , I believe the 2nd param is going to be the index, so just put the index as key rather than a potentially very long confusing string